include/boost/geometry/formulas/elliptic_meridian_arc_length.hpp

```diff
@@ -0,0 +1,214 @@
+        bool meridian;
+    };
+
+    template <typename T, typename Spheroid>
+    static result apply(T lon1, T lat1, T lon2, T lat2, Spheroid const& spheroid)
+    {
+        result res;
+
+        CT c0 = 0;
+        CT pi = math::pi<CT>();
+        CT half_pi = pi/CT(2);
+        CT diff = geometry::math::longitude_distance_signed<geometry::radian>(lon1, lon2);
+
+        if (lat1 > lat2)
+        {
+            std::swap(lat1, lat2);
+        }
+
+        if ( math::equals(diff, c0) ||
+            (math::equals(lat2, half_pi) && math::equals(lat1, -half_pi)) )
+        {
+            // single meridian not crossing pole
+            res.distance = apply(lat2, spheroid) - apply(lat1, spheroid);
+            res.meridian = true;
+        }
+
+        if (math::equals(math::abs(diff), pi))
+        {
+            // meridian crosses pole
+            CT lat_sign = 1;
+            if (lat1+lat2 < c0)
+            {
+                lat_sign = CT(-1);
+            }
+            res.distance = math::abs(lat_sign * CT(2) * apply(half_pi, spheroid)
+                               - apply(lat1, spheroid) - apply(lat2, spheroid));
+            res.meridian = true;
+        }
+        return res;
+    }
+
+    // Distance computation on meridians using series approximations
+    // to elliptic integrals. Formula to compute distance from lattitude 0 to lat
+    // https://en.wikipedia.org/wiki/Meridian_arc
+    // latitudes are assumed to be in radians and in [-pi/2,pi/2]
+    template <typename T, typename Spheroid>
+    static CT apply(T lat, Spheroid const& spheroid)
+    {
+        CT const a = get_radius<0>(spheroid);
+        CT const f = formula::flattening<CT>(spheroid);
+        CT n = f / (CT(2) - f);
+        CT M = a/(1+n);
+        CT C0 = 1;
+
+        if (Order == 0)
+        {
+           return M * C0 * lat;
+        }
+
+        CT C2 = -1.5 * n;
+
+        if (Order == 1)
+        {
+            return M * (C0 * lat + C2 * sin(2*lat));
+        }
+
+        CT n2 = n * n;
+        C0 += .25 * n2;
+        CT C4 = 0.9375 * n2;
+
+        if (Order == 2)
+        {
+            return M * (C0 * lat + C2 * sin(2*lat) + C4 * sin(4*lat));
+        }
+
+        CT n3 = n2 * n;
+        C2 += 0.1875 * n3;
+        CT C6 = -0.729166667 * n3;
+
+        if (Order == 3)
+        {
+            return M * (C0 * lat + C2 * sin(2*lat) + C4 * sin(4*lat)
+                      + C6 * sin(6*lat));
+        }
+
+        CT n4 = n2 * n2;
+        C4 -= 0.234375 * n4;
+        CT C8 = 0.615234375 * n4;
+
+        if (Order == 4)
+        {
+            return M * (C0 * lat + C2 * sin(2*lat) + C4 * sin(4*lat)
+                      + C6 * sin(6*lat) + C8 * sin(8*lat));
+        }
+
+        CT n5 = n4 * n;
+        C6 += 0.227864583 * n5;
+        CT C10 = -0.54140625 * n5;
+
+        // Order 5 or higher
+        return M * (C0 * lat + C2 * sin(2*lat) + C4 * sin(4*lat)
+                  + C6 * sin(6*lat) + C8 * sin(8*lat) + C10 * sin(10*lat));
+
```

Comment:
The Order 5 expression omits the terms

    n^4/16 * lat +  3*n^5/8 * sin(2*lat)

Also why aren't the coefficients given with full precision, e.g., replace

    0.729166667

with

    0.72916666666666667

or better yet by

    CT(35)/CT(48)

Finally, for full accuracy with the SRMmax ellipsoid (f = 1/150), the 6th-order
series should be used.  Why not include this too?

Reply:
Thank you for the comments. I updated the formula with the higher order terms.